corehq/form_processor/backends/couch.py

```diff
@@ -0,0 +1,78 @@
+from couchdbkit import ResourceNotFound
+
+from couchforms.models import doc_types, XFormInstance, XFormError
+from couchforms.exceptions import UnexpectedDeletedXForm
+from dimagi.utils.couch.undo import DELETED_SUFFIX
+
+from ..exceptions import XFormNotFound
+
+
+class XFormCouch(object):
+
+    @staticmethod
+    def create_from_generic(generic_xform, generic_attachment=None):
+        xform = XFormInstance.from_generic(generic_xform)
+        xform.save()
+        if generic_attachment:
+            xform.put_attachment(**generic_attachment.to_json())
+        return xform
+
+    @staticmethod
+    def get_attachment(xform_id, attachment_name):
+        return XFormInstance.get_db().fetch_attachment(xform_id, attachment_name)
+
+    @classmethod
+    def get_attachments(cls, xform_id):
+        xform = cls._get_xform(xform_id)
+        return xform.attachments
+
+    @classmethod
+    def archive(cls, xform_generic, user=None):
+        xform = cls._get_xform(xform_generic.id)
+        return xform.archive(user=user)
+
+    @classmethod
+    def unarchive(cls, xform_generic, user=None):
+        xform = cls._get_xform(xform_generic.id)
+        return xform.unarchive(user=user)
+
+    @classmethod
+    def get_xml_element(cls, xform_generic):
+        xform = cls._get_xform(xform_generic.id)
```

Comment:
this one also depends on the `_attachments` property which i had been assuming won't exist for the sql model